src/ctia/entity/asset_properties.clj

```diff
@@ -0,0 +1,134 @@
+  "new-asset-properties")
+
+(def-stored-schema StoredAssetProperties AssetProperties)
+
+(s/defschema PartialStoredAssetProperties
+  (csu/optional-keys-schema StoredAssetProperties))
+
+(def realize-asset-properties
+  (default-realize-fn "asset-properties" NewAssetProperties StoredAssetProperties))
+
+(def asset-properties-mapping
+  {"asset-properties"
+   {:dynamic false
+    :properties
+    (merge
+     em/base-entity-mapping
+     em/sourcable-entity-mapping
+     em/stored-entity-mapping
+     {:identity    em/tg-identity
+      :valid_time  em/valid-time
+      :asset_ref   em/token})}})
+
+(def-es-store AssetPropertiesStore :asset-properties StoredAssetProperties PartialStoredAssetProperties)
+
+(def asset-properties-fields
+  (concat sorting/default-entity-sort-fields
+          [:asset_ref]))
+
+(def asset-properties-sort-fields
+  (apply s/enum asset-properties-fields))
+
+(s/defschema AssetPropertiesFieldsParam
+  {(s/optional-key :fields) [asset-properties-sort-fields]})
+
+(s/defschema AssetPropertiesByExternalIdQueryParams
+  (st/merge
+   routes.common/PagingParams
+   AssetPropertiesFieldsParam))
+
+(def asset-properties-enumerable-fields
+  [])
+
+(def AssetPropertiesGetParams AssetPropertiesFieldsParam)
+
+(s/defschema AssetPropertiesSearchParams
+  (st/merge
+   routes.common/PagingParams
+   routes.common/BaseEntityFilterParams
+   routes.common/SourcableEntityFilterParams
+   AssetPropertiesFieldsParam
+   (st/optional-keys
+    {:query           s/Str
+     :sort_by         asset-properties-sort-fields})))
+
+(def asset-properties-histogram-fields
+  [:timestamp
+   :valid_time.start_time
+   :valid_time.end_time])
+
+(def asset-properties-routes
+  (entity-crud-routes
+   {:entity                   :asset-properties
+    :new-schema               NewAssetProperties
+    :entity-schema            AssetProperties
+    :get-schema               PartialAssetProperties
+    :get-params               AssetPropertiesGetParams
+    :list-schema              PartialAssetPropertiesList
+    :search-schema            PartialAssetPropertiesList
+    :external-id-q-params     AssetPropertiesByExternalIdQueryParams
+    :search-q-params          AssetPropertiesSearchParams
+    :new-spec                 :new-asset-properties/map
+    :realize-fn               realize-asset-properties
+    :get-capabilities         :read-asset-properties
+    :post-capabilities        :create-asset-properties
+    :put-capabilities         :create-asset-properties
+    :delete-capabilities      :delete-asset-properties
+    :search-capabilities      :search-asset-properties
+    :external-id-capabilities :read-asset-properties
+    :can-aggregate?           true
+    :histogram-fields         asset-properties-histogram-fields
+    :enumerable-fields        asset-properties-enumerable-fields
+    }))
+
+(def capabilities
+  #{:create-asset-properties
+    :read-asset-properties
+    :delete-asset-properties
+    :search-asset-properties})
```

Comment:
We have `asset`, `asset-mapping` and `asset-properties`. We define for each of this entities a set of capabilities. Their life cycle seems very correlated, I am wondering if we could not imagine having only `:read-asset`, `create-asset`, `delete-asset`, `search-asset` for all.
@gbuisson WDYT?

Reply:
No I'd rather not make an exception and have all the capabilities explicit, we will group them under the same root scope.